Sources/SwiftIfConfig/IfConfigEvaluation.swift

```diff
@@ -30,7 +30,8 @@ import SwiftSyntax
 ///   diagnose syntax errors in blocks where the check fails.
 func evaluateIfConfig(
   condition: ExprSyntax,
-  configuration: some BuildConfiguration
+  configuration: some BuildConfiguration,
+  outermostCondition: Bool = true
```

Comment:
Do we need to pass this around? Couldn’t you check `condition.keyPathInParent == \IfConfigClauseSyntax.condition`?

Reply:
It's more involved than checking the immediate parent, because we pass this through parentheses and prefix `!`. However, you're right that we can get this information from the syntax tree itself without passing it down as state. I've done that in https://github.com/swiftlang/swift-syntax/pull/2806